src/Expectation.php

```diff
@@ -522,10 +523,10 @@ public function toHaveKey($key, $value = null): Expectation
             $array = (array) $this->value;
         }
 
-        Assert::assertArrayHasKey($key, $array);
+        Assert::assertTrue(Arr::has($array, $key));
```

Comment:
As this assertions is now an `assertTrue`, the error message will not be as clear, maybe we should add a custom message?

Reply:
This could be resolved with 

`Assert::assertTrue(Arr::has($array, $key), "Failed asserting that an array has the key '$key'");`

but it will generate this error message:

> Failed asserting that an array has the key 'foo'
> Failed asserting that false is true.

Any idea on how to avoid the second part of the message? I think it is quite useless, but it is added from `Assert::assertTrue`

Reply:
I've implemented a possible workaround wrapping `Assert::assertTrue(Arr::has($array, $key));` in a try-catch block and generating a custom `ExpectationFailedException`:

```php
try {
    Assert::assertTrue(Arr::has($array, $key));
} catch (ExpectationFailedException $exception) {
    throw new ExpectationFailedException(
        "Failed asserting that an array has the key '$key'",
        $exception->getComparisonFailure()
    );
}
```